clients/stellar-relay-lib/src/connection/connector/message_reader.rs

```diff
@@ -43,39 +46,58 @@ pub(crate) async fn poll_messages_from_stellar(
 		}
 
 		// check for messages from Stellar Node.
-		let xdr = match read_message_from_stellar(&mut connector).await {
-			Err(e) => {
+		// if reading took too much time, flag it as "disconnected"
+		let xdr = match timeout(
+			Duration::from_secs(READ_TIMEOUT_IN_SECS),
+			read_message_from_stellar(&mut connector),
```

Comment:
Instead of having the timeout inside `read_message_from_stellar(...)`, I have it handled outside of it. This here is more understandable, than having in the `message_reader.rs`

Reply:
This is very nice improvement because we were reading from the stream [here](https://github.com/pendulum-chain/spacewalk/blob/7c7989875e95e1cfe3b0aeba9bb6af01d9e33e58/clients/stellar-relay-lib/src/connection/connector/message_reader.rs#L178) without a timeout.